configure.ac

```diff
@@ -4741,13 +4741,31 @@ esac
 
 case "$TARGET" in
 *32* | ARM | X86 | POWERPC | MIPS | SPARC | WASM | ARM6432)
+    AC_DEFINE([MONO_TARGET32], [1], [Target is 32bit])
+    AC_DEFINE([MONO_TARGET64], [0], [Target is not 64bits])
     target_sizeof_void_p=4
     ;;
 *64* | S390X)
+    AC_DEFINE([MONO_TARGET32], [0], [Target is not 32bit])
+    AC_DEFINE([MONO_TARGET64], [1], [Target is 64bits])
     target_sizeof_void_p=8
     ;;
 *)
-    AC_MSG_ERROR([unknown target])
+    AC_MSG_ERROR([Unable to determine if target is 32bit or 64bit.])
+    ;;
+esac
+
+case "$HOST" in
+*32* | ARM | X86 | POWERPC | MIPS | SPARC | WASM | ARM6432)
+    AC_DEFINE([MONO_HOST32], [1], [Host is 32bit])
+    AC_DEFINE([MONO_HOST64], [0], [Host is not 64bits])
+    ;;
+*64* | S390X)
+    AC_DEFINE([MONO_HOST32], [0], [Host is not 32bit])
+    AC_DEFINE([MONO_HOST64], [1], [Host is 64bits])
+    ;;
+*)
+    AC_MSG_ERROR([Unable to determine if host is 32bit or 64bit.])
```

Comment:
Is it ok for the default to be an error?
Or just remove all of this "for now"? (forever?)

Reply:
I lean slightly toward removing this.  Mostly because it's somewhat ambiguous what a "32-bit host" is.  I'm thinking of ARM64_32 for example.  Currently we're more specific (but somewhat - decentralized): we ask "is `sizeof(void*) == 8`" or about register sizes, or about alignment.  "64-bit host" could be shorthand for a number of conflated concerns and it might be difficult to untangle on a new architecture.

Reply:
Yeah I agree, let's wait until we have a good reason to use these defines. The important part of this PR are the printf warning fixes so let's get those in :)

Reply:
1. I pushed before seeing some of this, not all of it.
2. I cannot see yet what lambdageek pushed. Still looking.
3. Question is -- keep the MONO_TARGET stuff or not? Understood to remove MONO_HOST.

Reply:
Ok, while I cannot find the commit, the description is indicatative:
```
Pushed a commit to remove MONO_{HOST,TARGET}{32,64}
```
so I'll do what it says.

Reply:
Though I think the `MONO_TARGET{32,64}` stuff is reasonable, and CoreCLR has similar to some/all of this.